src/mono/mono/metadata/icall.c

```diff
@@ -2525,6 +2525,23 @@ ves_icall_RuntimeType_GetInterfaces (MonoReflectionTypeHandle ref_type, MonoErro
 	return NULL_HANDLE_ARRAY;
 }
 
+static gboolean
+method_is_reabstracted (MonoMethod *method)
+{
+	/* only on interfaces */
+	/* method is marked "final abstract" */
+	/* FIXME: we need some other way to detect reabstracted methods.  "final" is an incidental detail of the spec. */
+	return method->flags & METHOD_ATTRIBUTE_FINAL && method->flags & METHOD_ATTRIBUTE_ABSTRACT;
+}
+
+static gboolean
+method_is_dim (MonoMethod *method)
+{
+	/* only valid on interface methods*/
+	/* method is marked "virtual" but not "virtual abstract" */
+	return method->flags & method->flags & METHOD_ATTRIBUTE_VIRTUAL && !(method->flags & METHOD_ATTRIBUTE_ABSTRACT);
+}
```

Comment:
This looks wrong.

Reply:
There are some accessors like m_method_is_virtual () etc.

Reply:
> This looks wrong.

@vargaz do you mean not using the accessor is wrong, or checking for a default interface method using the method flags is wrong?

Reply:
This part:
method->flags & method->flags & METHOD_ATTRIBUTE_VIRTUAL